### PR TITLE
Split test categories for IoT hub service and IoT hub client

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: iothub/device/tests/RetryPolicies/IotHubClientExponentialBackoffRetryPolicyTests.cs [hunk 1]
```diff
@@ -2,15 +2,13 @@
...
 using System;
-using System.Threading;
 using FluentAssertions;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 
 namespace Microsoft.Azure.Devices.Client.Test
 {
     [TestClass]
     [TestCategory("Unit")]
-    [TestCategory("IoTHub")]
```
Comment (abhipsaMisra) [inline iothub/device/tests/RetryPolicies/IotHubClientExponentialBackoffRetryPolicyTests.cs:13]: I don't see the advantage of tagging unit tests as IoTHub. They have minimal running time and don't provide any substantial benefit if we exclude them from our runs.

**Review by tmahmood-microsoft — APPROVED** [review]

File: iothub/device/tests/RetryPolicies/IotHubClientExponentialBackoffRetryPolicyTests.cs (hunk 1 above)
Comment (drwill-ms) [reply iothub/device/tests/RetryPolicies/IotHubClientExponentialBackoffRetryPolicyTests.cs:13]: The goal was to separate hub from dps tests.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**